test/blas.cpp

```diff
@@ -383,6 +384,48 @@ float h_gold_batch[18] = {30.f, 84.f, 138.f,
                           81.f, 36.f, 87.f,
                           69.f, 30.f, 69.f};
 
+#ifndef AF_CPU
+TEST(MatrixMultiply, half) {
+  SUPPORTED_TYPE_CHECK(af_half);
+
+  array A32 = array(3, 3, f32);
+  A32.write<float>((float*)h_lhs, sizeof(float) * 9);
+  array A16 = A32.as(f16);
+  af_print(A16);
+
+  array B32 = array(3, 3, f32);
+  B32.write<float>((float*)h_rhs, sizeof(float) * 9);
+  array B16 = B32.as(f16);
+  af_print(B16);
+
+  af_array C32 = 0;
+  const float alpha32 = 1.0f;
+  const float beta32 = 0.0f;
+  af_gemm(&C32, AF_MAT_NONE, AF_MAT_NONE, &alpha32, A32.get(), B32.get(), &beta32);
+  af_print_array(C32);
+  array expected32 = array(3, 3, h_gold);
+  af_print(expected32);
+  ASSERT_ARRAYS_NEAR(expected32, af::array(C32), 0.001);
+
+  array expected16 = expected32.as(f16);
+  af_print(expected16);
+  {
+    af_array C16 = 0;
+    const af_half alpha16 = {0x03c00}; // 1.0 : 0 01111 0000000000
+    const af_half beta16 = {0x00000}; //  0.0 : 0 00000 0000000000
+    af_gemm(&C16, AF_MAT_NONE, AF_MAT_NONE, &alpha16, A16.get(), B16.get(), &beta16);
```

Comment:
would be better to split this into a separate test.

Reply:
done